### PR TITLE
Better logging in case of failed ClusterConfig updates

### PR DESCRIPTION
**Pull request by lennartkoopmann**
Before this change it was really hard to figure out why a ClusterConfig update would fail. Especially in case of a key mismatch between the web config and the ClusterConfig class, there was no way that out.

### CODE REVIEW

**Comment by kroepke** [comment]
If this is targeted for 2.1, I think the PR should be against master. I'm trying to keep the released branches as lean as possible.

File: graylog2-server/src/main/java/org/graylog2/rest/resources/system/ClusterConfigResource.java [hunk 1]
```diff
@@ -107,10 +112,14 @@ public Response update(@ApiParam(name = "configClass", value = "The name of the
...
-        clusterConfigService.write(o);
-
-        return Response.accepted(o).build();
+        try {
+            final Object o = objectMapper.readValue(body, cls);
+            clusterConfigService.write(o);
+            return Response.accepted(o).build();
+        }catch(Exception e) {
+            LOG.error("Could not write new cluster config.", e);
+            return Response.status(400).build();
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/ClusterConfigResource.java]: The response should contain some text about the nature of the problem.

File: graylog2-server/src/main/java/org/graylog2/rest/resources/system/ClusterConfigResource.java [hunk 2]
```diff
@@ -107,10 +112,14 @@ public Response update(@ApiParam(name = "configClass", value = "The name of the
...
 
-        final Object o = objectMapper.readValue(body, cls);
-        clusterConfigService.write(o);
-
-        return Response.accepted(o).build();
+        try {
+            final Object o = objectMapper.readValue(body, cls);
+            clusterConfigService.write(o);
+            return Response.accepted(o).build();
+        }catch(Exception e) {
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/ClusterConfigResource.java]: Missing whitespace. :wink:

File: graylog2-server/src/main/java/org/graylog2/rest/resources/system/ClusterConfigResource.java (hunk 1 above)
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/ClusterConfigResource.java]: Maybe simply throw a `WebApplicationException` here and add the original exception for its stack trace.

File: graylog2-server/src/main/java/org/graylog2/rest/resources/system/ClusterConfigResource.java [hunk 3]
```diff
@@ -58,6 +60,9 @@
 @Path("/system/cluster_config")
 @Produces(MediaType.APPLICATION_JSON)
 public class ClusterConfigResource extends RestResource {
+    private static final Logger LOG = LoggerFactory.getLogger(ClusterConfigResource.class);
+
+
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/ClusterConfigResource.java]: Useless newline.

**Pull request closed without merging**